### PR TITLE
[New Model]: DemandAndCapacityNotification 1.0.0

### LINKED ISSUE #527: [New Model]: Notifications 1.0.0
## Model Description

The Notification model allows the exchange of alerts between business partners. Alerts can occur on site(s) (e.g. strike, earthquake, fire, flood) which can have an effect on another site(s) (demand-reduction, capacity-reduction, capacity-increase, demand-increase). This model also allows to exchange the affected item numbers and a custom user message.

## Relevant Standards

There are no relevant standards available, however, this standard will be reusing some of existing elements from other Catena-X standards such as relevant inventory item identifiers.

## Example Data

I included the "header" in the following example because it has properties which are relevant for the functionality. The actual model example is in the content.
```json
{
    "header": {
        "context": "RES-PURIS-TRACE-DCM-Alert:1.0.0",
        "expectedResponseBy": "2023‐06‐21T21:24:00+07:00",
        "messageId": "3b4edc05-e214-47a1-b0c2-1d831cdd9ba9",
        "relatedMessageId": "d9452f24-3bf3-4134-b3eb-68858f1b2362",
        "recipientBpn": "BPNL6666787765VQ",
        "senderBpn": "BPNL7588787849VQ",
        "sentDateTime": "2023‐06‐19T21:24:00+07:00",
        "version": "urn:samm:io.catenax.message_header:1.0"
    },
    "content": {
        "notifications": [
            {
              "affectedSitesSender": [
                "BPNS7588787849VQ"
              ],
              "affectedSitesReceiver": [
                "BPNS6666787765VQ"
              ], 
              "startDateOfEffect": "2023-12-13T15:00:00+01:00",
              "materialNumberSupplier": [
                "MNR-8101-ID146955.001"
              ],
              "materialNumberCustomer": [
                "MNR-7307-AU340474.002"
              ],
              "materialNumberDigitalTwin": [
                "f81d4fae-7dec-11d0-a765-00a0c91e6bf6"
              ],
              "leadingRootCause": "strike",
              "relatedSourceMessageId": "urn:uuid:b3cdDeA6-DC8a-0D7f-dfb5-A6B74bEA2f13",
              "effect": "demand-reduction",
              "text": "HelloWorld",
              "expectedEndDateOfEffect": "2023-12-17T08:00:00+01:00",
              "status": "resolved",
              "contentChangedAt": "2024-01-15T12:32:57.450+01:00",
            }
        ]
    }
}
```

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by stiwan8020**
## Description

The Notification model allows the exchange of alerts between business partners. Alerts can occur on site(s) (e.g. strike, earthquake, fire, flood) which can have an effect on another site(s) (demand-reduction, capacity-reduction, capacity-increase, demand-increase). This model also allows to exchange the affected item numbers and a custom user message.

 -->

Closes #527

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by tom-rm-meyer-ISST** [comment]
Checked validation of model. For some reason the shared aspect for bpnl is lookup up in the ttl files directory. Locally running the cli version 2.5.1 works.

**Comment by stiwan8020** [comment]
@tom-rm-meyer-ISST problem solved: the namespace of the model was wrong 🤦‍♂️
- @prefix : <urn:samm:org.eclipse.notification:1.0.0#>.
+ @prefix : <urn:samm:io.catenax.notifications:1.0.0#> .

File: io.catenax.notifications/1.0.0/Notifications.ttl [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+#######################################################################
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.notifications:1.0.0#> .
+@prefix ext-number: <urn:samm:io.catenax.shared.business_partner_number:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.notifications/1.0.0/Notifications.ttl:26]: please use business_partner_number 2.0.0 as it is already released.

File: io.catenax.notifications/1.0.0/Notifications.ttl [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+@prefix ext-uuid: <urn:samm:io.catenax.shared.uuid:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.notifications/1.0.0/Notifications.ttl:27]: here also use uuid2.0.0 model. I have recently released new shared model versions.
Its always better to use the latest models.

File: io.catenax.notifications/1.0.0/Notifications.ttl [hunk 3]
```diff
@@ -0,0 +1,177 @@
+#######################################################################
+# Copyright (c) 2023-2024 ISTOS GmbH (a member of the DMG Mori Group)
```
Comment (agg3fe) [inline io.catenax.notifications/1.0.0/Notifications.ttl:2]: only 2024 is required

File: io.catenax.notifications/1.0.0/Notifications.ttl [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+:Notifications a samm:Aspect ;
+   samm:preferredName "Notifications"@en ;
+   samm:description "This aspect model defines the standard for notification exchange between business partners. Notifications refer to specific events which affect production or demand."@en ;
+   samm:properties ( [ samm:property :affectedSitesSender; samm:optional true ] [ samm:property :affectedSitesRecipient; samm:optional true ] :leadingRootCause :effect [ samm:property :text; samm:optional true ] [ samm:property :materialNumberDigitalTwin; samm:optional true ] :startDateOfEffect [ samm:property :expectedEndDateOfEffect; samm:optional true ] :status :contentChangedAt [ samm:property :relatedSourceMessageId; samm:optional true ] [ samm:property :manufacturerPartId; samm:optional true ] [ samm:property :customerPartId; samm:optional true ] ) ;
+   samm:operations ( ) ;
+   samm:events ( ) .
+
+:affectedSitesSender a samm:Property ;
+   samm:preferredName "Affected sites of the sender."@en ;
+   samm:description """The affected Business Partner SITE Numbers of the sending party.
```
Comment (agg3fe) [inline io.catenax.notifications/1.0.0/Notifications.ttl:38]: There is unnecessary space in the description. please correct this.
Correct in other places also.

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.notifications/RELEASE_NOTES.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+All notable changes to this model will be documented in this file.
+
+## [1.0.0]
```
Comment (agg3fe) [inline io.catenax.notifications/RELEASE_NOTES.md:4]: please add date 2024-02-05, if you want MS3 on next meeting.

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by stiwan8020** [comment]
@agg3fe after a lot of discussions we finally agreed on a common model name. Now the model is ready for the final review

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by agg3fe** [comment]
property name change required.
@jacewski-bosch

**Pull request merged**